### ISSUE TITLE
GOES-XRS tests need tidying and updating

### ISSUE THREAD

**Issue by hayesla** [sunpy/sunpy#6457]
### Provide a general description of the issue or problem.

The `XRSTimeSeries` supports a range of files and the testing suite to test these is a bit all over the place and doesn't fully test all possible file paths. 

The files that XRSTimeSeries can read are:

1. GOES-R XRS 1s flux (netCDF files)
      - GOES 16 files
      - GOES 17 files
2. GOES-R XRS 1m averages (netCDF files)
      - GOES 16 files
      - GOES 17 files
3. GOES XRS 2s fluxes (netCDF files)
      - GOES 13 files
      - GOES 14 files
      - GOES 15 files
4. GOES XRS 1m averages (netCDF files)
      - GOES 13 files
      - GOES 14 files
      - GOES 15 files
5. GOES XRS 2s fluxes (FITS files)
      - GOES 1-15 files

So basically 5 different file types that need to be tested. Most are, but I think a tidier way of doing this would be good.

**Comment by aritrasinha108** [comment]
@hayesla @nabobalis. I looked at the file [`sunpy.timeseries.sources.tests.test_goes.py`](https://github.com/sunpy/sunpy/blob/main/sunpy/timeseries/sources/tests/test_goes.py). And it does have a lot of tests which do not cover all the file paths. Do you have any specific suggestion here on how to reorganize the whole thing? Oris it fine if we just add the filepaths into the tests in which they are missing.

**Comment by nabobalis** [comment]
> @hayesla @nabobalis. I looked at the file [`sunpy.timeseries.sources.tests.test_goes.py`](https://github.com/sunpy/sunpy/blob/main/sunpy/timeseries/sources/tests/test_goes.py?rgh-link-date=2023-03-13T23%3A36%3A08Z). And it does have a lot of tests which do not cover all the file paths. Do you have any specific suggestion here on how to reorganize the whole thing? Oris it fine if we just add the filepaths into the tests in which they are missing.

I have no ideas as I've not really looked at the file. But parametrizing the files as input to each test function might be the place to start?

**Comment by aritrasinha108** [comment]
> > @hayesla @nabobalis. I looked at the file [`sunpy.timeseries.sources.tests.test_goes.py`](https://github.com/sunpy/sunpy/blob/main/sunpy/timeseries/sources/tests/test_goes.py?rgh-link-date=2023-03-13T23%3A36%3A08Z). And it does have a lot of tests which do not cover all the file paths. Do you have any specific suggestion here on how to reorganize the whole thing? Oris it fine if we just add the filepaths into the tests in which they are missing.
> 
> I have no ideas as I've not really looked at the file. But parametrizing the files as input to each test function might be the place to start?

Actually, most of the tests are written by parametrizing the files only. If it's possible could you please help me finding any scope of improvement here. One thing I could find is a few test don't cover all the applicable files. I am working on fixing that, but other than that I couldn't find anything.

**Comment by nabobalis** [comment]
Then in that case, just add more files to make sure we cover all of them and that will be enough for this issue.
